cpp/src/slice2cs/CsUtil.cpp

```diff
@@ -31,21 +31,21 @@ namespace
         // Keyword list. *Must* be kept in alphabetical order.
         //
         static const string keywordList[] = {
-            "abstract", "as",        "async",     "await",      "base",      "bool",     "break",    "byte",
-            "case",     "catch",     "char",      "checked",    "class",     "const",    "continue", "decimal",
-            "default",  "delegate",  "do",        "double",     "else",      "enum",     "event",    "explicit",
-            "extern",   "false",     "finally",   "fixed",      "float",     "for",      "foreach",  "goto",
-            "if",       "implicit",  "in",        "int",        "interface", "internal", "is",       "lock",
-            "long",     "namespace", "new",       "null",       "object",    "operator", "out",      "override",
-            "params",   "private",   "protected", "public",     "readonly",  "ref",      "return",   "sbyte",
-            "sealed",   "short",     "sizeof",    "stackalloc", "static",    "string",   "struct",   "switch",
-            "this",     "throw",     "true",      "try",        "typeof",    "uint",     "ulong",    "unchecked",
-            "unsafe",   "ushort",    "using",     "virtual",    "void",      "volatile", "while"};
+            "abstract", "as",        "async",     "await",   "base",      "bool",     "break",      "byte",
+            "case",     "catch",     "char",      "checked", "class",     "const",    "continue",   "decimal",
+            "default",  "delegate",  "do",        "double",  "else",      "enum",     "event",      "explicit",
+            "extern",   "false",     "finally",   "fixed",   "float",     "for",      "foreach",    "goto",
+            "if",       "implicit",  "in",        "int",     "interface", "internal", "is",         "lock",
+            "long",     "namespace", "new",       "null",    "object",    "operator", "out",        "override",
+            "params",   "private",   "protected", "public",  "readonly",  "record",   "ref",        "required",
+            "return",   "sbyte",     "scoped",    "sealed",  "short",     "sizeof",   "stackalloc", "static",
+            "string",   "struct",    "switch",    "this",    "throw",     "true",     "try",        "typeof",
+            "uint",     "ulong",     "unchecked", "unsafe",  "ushort",    "using",    "var",        "virtual",
+            "void",     "volatile",  "while"};
         bool found = binary_search(
             &keywordList[0],
             &keywordList[sizeof(keywordList) / sizeof(*keywordList)],
-            name,
-            Slice::CICompare());
```

Comment:
There are no uppercase keywords in C#, so there's no need to pass our custom comparator.
In the other languages, we only pass it when it's necessary.